mantidproject/mantid
# Add MatrixWorkspaceDisplay to the Workbench

**Description of work.**
Returns the `Show Data` functionality to the Workbench with the MatrixWorkspaceDisplay. Tests added.

Too many plots confirmation dialogue has been changed to Yes/No instead of Confirm/Cancel

Added `maskedBinsIndices` to `MatrixWorkspace`, and exposed it to Python, along with `hasMaskedBins`. Tests added.

Created and exposed to Python new WorkspaceCreationHelper method - `create2DWorkspace123WithMaskedBin`. Tests added.

**Using matplotlib with Qt5**
If you encounter any matplotlib issues when plotting, try upgrading it with the following commands:

`python -m pip install matplotlib -U --no-deps`
`python -m pip install kiwisolver -U --no-deps`
`python -m pip install backports.functools_lru_cache -U --no-deps`

**To test:**
1. Load a workspace
2. Double click on a workspace, or right click -> `Show Data`

That should open the MatrixWorkspaceDisplay. Inside it only the `View Detector Tables` functionality has not been added. Everything else should be exactly the same as the MantidPlot version.

- Open the same WS in both Workbench and MantidPlot and spot any difference in behaviours.
- Try loading not-matrix workspaces into it. Compare what happens (error, crash?) versus what MantidPlot does.

Fixes #21593 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
